stdlib/3/os/__init__.pyi

```diff
@@ -285,63 +356,133 @@ def fstat(fd: int) -> stat_result: ...
 def ttyname(fd: int) -> str: ...  # Unix only
 def write(fd: int, string: bytes) -> int: ...
-def access(path: _PathType, mode: int) -> bool: ...
-def chdir(path: _PathType) -> None: ...
+if sys.version_info >= (3, 3):
+    def access(path: _FdOrPathType, mode: int, *, dir_fd: Optional[int] = ...,
+               effective_ids: Optional[bool] = ..., follow_symlinks: Optional[bool] = ...) -> bool: ...
+else:
+    def access(path: _PathType, mode: int) -> bool: ...
+def chdir(path: _FdOrPathType) -> None: ...
 def fchdir(fd: int) -> None: ...
 def getcwd() -> str: ...
 def getcwdb() -> bytes: ...
-def chflags(path: _PathType, flags: int) -> None: ...  # Unix only
+if sys.version_info >= (3, 3):
+    def chflags(path: _PathType, flags: int, *, follow_symlinks: Optional[bool] = ...) -> None: ...  # Unix only
+    def chmod(path: _FdOrPathType, mode: int, *, dir_fd: Optional[int] = ..., follow_symlinks: Optional[bool] = ...) -> None: ...
+    def chown(path: _FdOrPathType, uid: int, gid: int, *, dir_fd: Optional[int] = ..., follow_symlinks: Optional[bool] = ...) -> None: ...  # Unix only
+else:
+    def chflags(path: _PathType, flags: int) -> None: ...  # Unix only
+    def chmod(path: _PathType, mode: int) -> None: ...
+    def chown(path: _PathType, uid: int, gid: int) -> None: ...  # Unix only
 def chroot(path: _PathType) -> None: ...  # Unix only
-def chmod(path: _PathType, mode: int) -> None: ...
-def chown(path: _PathType, uid: int, gid: int) -> None: ...  # Unix only
 def lchflags(path: _PathType, flags: int) -> None: ...  # Unix only
 def lchmod(path: _PathType, mode: int) -> None: ...  # Unix only
 def lchown(path: _PathType, uid: int, gid: int) -> None: ...  # Unix only
-def link(src: _PathType, link_name: _PathType) -> None: ...
+if sys.version_info >= (3, 3):
+    def link(src: _PathType, link_name: _PathType, *, src_dir_fd: Optional[int] = ...,
+             dst_dir_fd: Optional[int] = ..., follow_symlinks: Optional[bool] = ...) -> None: ...
+else:
+    def link(src: _PathType, link_name: _PathType) -> None: ...
 
-@overload
-def listdir(path: str = ...) -> List[str]: ...
-@overload
-def listdir(path: bytes) -> List[bytes]: ...
+if sys.version_info >= (3, 3):
+    @overload
+    def listdir(path: Optional[str] = ...) -> List[str]: ...
+    @overload
+    def listdir(path: bytes) -> List[bytes]: ...
+    @overload
+    def listdir(path: int) -> List[str]: ...
+else:
+    @overload
+    def listdir(path: Optional[str] = ...) -> List[str]: ...
+    @overload
+    def listdir(path: bytes) -> List[bytes]: ...
 
-def lstat(path: _PathType) -> stat_result: ...
-def mkfifo(path: _PathType, mode: int = ...) -> None: ...  # Unix only
-def mknod(filename: _PathType, mode: int = ..., device: int = ...) -> None: ...
+if sys.version_info >= (3, 3):
+    def lstat(path: _PathType, *, dir_fd: Optional[int] = ...) -> stat_result: ...
+    def mkdir(path: _PathType, mode: Optional[int] = ..., *, dir_fd: Optional[int] = ...) -> None: ...
```

Comment:
These mode arguments are not Optional. More instances of the same below: mode arguments should be int, dir_fd is correct as Optional[int], and all the boolean args should just be `bool`.